includes/civicrm.shortcodes.php

```diff
@@ -752,7 +752,22 @@ public function preprocess_atts($atts) {
         if ($mode == 'preview' || $mode == 'test') {
           $args['action'] = 'preview';
         }
-        $args['q'] = 'civicrm/contribute/transact';
+
+        switch ($action) {
+          case 'transact':
+            $args['q'] = 'civicrm/contribute/transact';
+            break;
+
+          case 'setup':
```

Comment:
The Direct URL is http://wpcvmaster.test/civicrm/contribute/campaign/?action=add&reset=1&pageId=1&component=contribute   The shortcode we get is `[civicrm component="contribution" id="1" action="setup" mode="live" hijack="0"]`  does the ID need to get mapped to `PageId`

Reply:
Fixed in #253